### PR TITLE
Update: Transdecoder to 5.5.0

### PR DESCRIPTION
**Pull request by stephanflemming**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

GitHub: https://github.com/TransDecoder/TransDecoder
GitHubWiki: https://github.com/TransDecoder/TransDecoder/wiki
Google User Group: https://groups.google.com/forum/#!forum/transdecoder-users
Conda: https://anaconda.org/bioconda/transdecoder
Conda recipe: https://bioconda.github.io/recipes/transdecoder/README.html
(My) parameter overview: https://docs.google.com/spreadsheets/d/1ancYJDZtri9hAizVgDwLwgCO-GVashAYusYmVBWi0h4/edit#gid=0

Fix: https://github.com/galaxyproject/tools-iuc/issues/2626

- most parameters changed since last wrapper update
- replaced test data file, because TransDecoder.Predict failed with the old one
- didn't split TransDecoder.LongOrfs and TransDecoder.Predict into two wrappers, because the second cannot run without the first
- therefore TransDecoder.Predict is optional, but activated by default, because this step is not required to generate files for the (external) homology search

@j-batista @abretaud maybe you want to have a look

### CODE REVIEW

**Comment by stephanflemming** [comment]
Not sure why this test fails.

**Comment by bgruening** [comment]
Triggered a rerun! Thanks.

File: tools/transdecoder/transdecoder.xml [hunk 1]
```diff
@@ -1,134 +1,420 @@
...
+                            <option value="dynamic" selected="true">Dynamic</option>
+                            <option value="strict">Strict</option>
+                        </param>
+                        <when value="dynamic"/>
+                        <when value="strict">
+                            <param argument="--retain_long_orfs_length" type="integer" value="1000000" min="0" label="Set long ORFs length" help="Retain all ORFs found that are equal or longer than these many nucleotides even if no other evidence marks it as coding."/>
+                        </when>
+                    </conditional>
+                    <param argument="--retain_blastp_hits" type="data" format="tabular" optional="true" label="Select BlastP result file" help="Any ORF with a blast match will be retained in the final output. (outfmt 6 format)"/>
+                    <param argument="--retain_pfam_hits" type="data" format="fasta" optional="true" label="Select Pfam result file" help=" Domain table output file from running hmmscan to search Pfam. Any ORF with a pfam domain hit will be retained in the final output. (domtblout file)"/>
```
Comment (abretaud) [inline tools/transdecoder/transdecoder.xml:110]: Is it really `fasta` datatype?

**Review by abretaud — APPROVED** [review]
LGTM, thanks @stephanflemming!
Just a little question

File: tools/transdecoder/transdecoder.xml (hunk 1 above)
Comment (stephanflemming) [reply tools/transdecoder/transdecoder.xml:110]: Good find! Couldn't test this parameter. A little search gives some exampe files in other projects. It seems to be tabular.

https://github.com/arendsee/rhmmer/blob/master/tests/testthat/sample-data/five/domtblout

**Comment by abretaud** [comment]
Thanks for the change!
Do you want another review (@j-batista ?) or do I merge now?

**Comment by stephanflemming** [comment]
From my site, it can be merged.

**Pull request merged**